includes/Admin/Editor.php

```diff
@@ -344,11 +344,11 @@ public function get_editor_settings(): array {
 		$user        = wp_get_current_user();
 
 		/** This filter is documented in wp-admin/includes/post.php */
-		$show_locked_dialog = apply_filters( 'show_post_locked_dialog', true, $post, $user );
-		$nonce              = wp_create_nonce( 'wp_rest' );
-		$mime_types         = $this->types->get_allowed_mime_types();
-		$image_mime_types   = $this->types->get_allowed_image_mime_types();
-		$audio_mime_types   = $this->types->get_allowed_audio_mime_types();
+		$show_locked_dialog     = apply_filters( 'show_post_locked_dialog', true, $post, $user );
+		$nonce                  = wp_create_nonce( 'wp_rest' );
+		$mime_types             = $this->types->get_allowed_mime_types();
+		$mime_types['ws_image'] = $this->types->get_allowed_image_mime_types();
```

Comment:
What is `ws_image` supposed to mean?

Reply:
Web safe or web story. Not decided yet. 😄 

ws_image and image mime are different. web site images do not include SVGs.

Reply:
Ah, so this is referring to things like poster images and publisher logos? What currently `allowedImageMimeTypes` is used for?

I see. So the structure is this:

```
'allowedMimeTypes' = [
  'image' => ..., // From get_allowed_mime_types (), for the image element
  'video' => ..., // From get_allowed_mime_types (), for the video element
  'audio' => ..., // // From get_allowed_mime_types (), for the audio element should we ever have one, but overridden to be used for bg audio
  'ws_image' => ... // For poster images, publisher logo, etc.
]
```

That's not very intuitive at all then and just as confusing as what we currently have.

Why not something like this

```
'allowedMimeTypes' = [
  'elements' => [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ],
  'story' => [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ]
]
```

Or flatter:

```
'allowedElementMimeTypes' = [
  'image' => ...,
  'video' => ...,
  'audio' => ...,
],
'allowedStoryMimeTypes' = [
  'image' => ...,
  'video' => ...,
  'audio' => ...,
]
```

Reply:
Okay, but that is very long. It also doesn't take into account pages. So media can embed / linked to elements, pages and story levels. So would you want 

```
'allowedMimeTypes' = [
  'elements' => [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ],
  'page' => [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ]
  'story' => [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ]
]
```

Reply:
I don't think we need 3 different ones 🤔 

Maybe this with different names:

```
'allowedMimeTypes' = [
  'story' => [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ],
  'elements' => [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ]
]
```

Reply:
So with your suggestion I think right now it would only be like this, right?

```
'allowedMimeTypes' = [
  'elements' => [
    'image' => ...,
    'video' => ...,
  ],
  'story' => [
    'audio' => ..., // Bg audio
  ]
  'metadata' => [
    'image' => ..., // Poster images
  ]
]
```

Reply:
Something like this 

```
'allowedMimeTypes' = [
  'elements' => [
    'image' => ...,
    'video' => ...,
  ],
  'pages' => [
    'audio' => ..., // Bg audio page level
  ]
  'metadata' => [
    'image' => ..., // Poster images
    'audio' => ..., // Bg audio story level
  ]
]
```

Reply:
To be honest, I would go back to 
```
'allowedMimeTypes' = [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
  ],
```

This web safe image thing, is only really an issue if you enable SVG support. We could have something hardcoded to remove svg from allow list, in javascript. it is only for posters and publisher logos. 

We could also do this. 

```
'allowedMimeTypes' = [
    'image' => ...,
    'video' => ...,
    'audio' => ...,
    'vector' => ['image/svg'],
  ],
```

Then we could keep image array, "clean" with only web safe mime types in the images.

Reply:
There is no need to separate story-level and page-level bg audio.

> This web safe image thing, is only really an issue if you enable SVG support.

SVG would fall under `elements.image`, I don't see why this would need to be a separate thing.

Reply:
I don't think separating into story / metadata / element makes any sense. It makes it confusing, which we are trying to get away from. 

SVGs can only be used in one context, which is image elements, all other contexts, like poster, publisher logo etc, it's does make any sense. By having image/svg in another key, it means for the contexts that support vector images, you could concat image and vector arrays together. For all other contexts, that require images that embeddable / web safe like posters, you would use images.  

We could also just not worry about in this PR and follow up when we officially support svgs.

Reply:
> SVGs can only be used in one context, which is image elements

Which is why it should not be a separate `vectors` list.

> I don't think separating into story / metadata / element makes any sense. It makes it confusing, which we are trying to get away from.

The current `ws_image` thing is way more confusing.

Another idea off the top of my head:

```
'allowedMimeTypes' = [
  'elements' => [
    'image' => ...,
    'video' => ...,
   // ... whatever new element types we might have in the future, e.g. audio elements
  ],
  'backgroundAudio' => ...,
  'storyPoster' => ...,
  'publisherLogo' => ...
],
```

Similarly, in the dashboard, for consistency:

```
'allowedMimeTypes' = [
  'publisherLogo' => ...
],
```